frontend/lib/Repository/UserRepository/UserGateway.dart

```diff
@@ -1,28 +1,45 @@
+import 'package:adpv_frontend/DataModels/User/auth_token_response.dart';
 import 'package:adpv_frontend/DataModels/User/user.dart';
-import 'package:adpv_frontend/DataModels/User/user_role.dart';
+
 import 'package:adpv_frontend/Repository/UserRepository/AuthGateway.dart';
 
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+
 import '../../Views/loginViewUtils.dart';
 
+const accessKey = 'access';
+const refreshKey = 'refresh';
 class UserGateway {
   AuthGetaway authGetaway = AuthGetaway();
+  FlutterSecureStorage secureStorage = const FlutterSecureStorage();
   late User user;
 
-  bool isAdmin() => user.userRole.contains(UserRole.admin);
+  bool isAdmin() => user.userRoles.contains(UserRole.admin);
 
-  void refreshToken() {
-    authGetaway
+  void refreshToken() async {
+    await authGetaway
         .refreshToken()
-        .then((value) => user.tokenResponse = value.tokens!);
+        .then((value) => user = User(value.tokens!));
   }
 
   Future<AuthResponse> authenticateUser(AuthenticateForm form) async {
-    final AuthResponse response = await authGetaway.authenticateUser(form);
-    if (response.success) {
-      user = User(response.tokens!);
+    final String? access = await secureStorage.read(key: accessKey);
+    final String? refresh = await secureStorage.read( key: refreshKey);
+
+    if (access != null && refresh != null) {
```

Comment:
Zastanawiam się czy skoro mamy już token'y to czy nie powinniśmy pominąć ekranu logowania? Bo aktualnie mamy tak, że pomimo faktu, że je mamy, to użytkownik nadal widzi ekran logowania, a chyba nie powinno tak być. Trzeba się też zastanowić, co w przypadku, gdy tokeny się przedawnią

Reply:
inny task

Reply:
To w takim razie nie jest potrzebne ściąganie token'a z cash'a w gateway'u, w przypadku kiedy on istnieje, bo skoro się logujemy, to chcemy przypisać sobie nowy token, a nie korzystać ze starego